Squares.Infrastructure/Repositories/SquaresDetectionEntriesRepository.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Squares.Application.Interfaces.Repositories;
+using Squares.Domain.Entities;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Squares.Infrastructure.Repositories
+{
+    /// <summary>
+    ///  Dummy implementation of persistance layer
+    /// </summary>
+    public class SquaresDetectionEntriesRepository : ISquaresDetectionEntriesRepository
+    {
+
+        public ConcurrentDictionary<string, SquaresDetectionEntry> IdentifiedSquares;
+        public SquaresDetectionEntriesRepository() 
+        {
+            this.IdentifiedSquares = new ConcurrentDictionary<string, SquaresDetectionEntry>();
+        }
+
+        public void AddAsync(SquaresDetectionEntry entity)
+        {
+            IdentifiedSquares.TryAdd(entity.Id, entity);
+        }
+
+        public void DeleteAsync(string id)
+        {
+            IdentifiedSquares.TryRemove(id, out SquaresDetectionEntry entity);
+        }
+
+        public IList<SquaresDetectionEntry> GetAllAsync()
```

Comment:
Optional (or Consider): Might be a good idea to consider pagination, since the retrieval of data could be too big to handle smoothly. This would make he application more scalable.